src/core/Database.cpp

```diff
@@ -199,6 +201,30 @@ Group* Database::findGroupRecursive(const Uuid& uuid, Group* group)
     return nullptr;
 }
 
+namespace {
+    QList<Entry*> resolveReferencesRecursive(const Uuid& uuid, const Group* group)
```

Comment:
This function should be a member of the Database class. Also recommend overloading the Database::resolveReferences function with this one accepting a specific group.